src/local-history/local-history-manager.ts

```diff
@@ -75,93 +72,117 @@ export class LocalHistoryManager {
-                });
+                        // Get the file system path for the selection.
+                        const selectionFsPath = path.join(`${hashedFolderPath}`, selection.description!);
+
+                        // Show the diff between the active editor and the selected local history file.
+                        this.displayDiff(vscode.Uri.file(selectionFsPath), textEditor.document.uri);
+                    });
+            });
         }
         else {
             vscode.window.showInformationMessage('File does not belong to a workspace. Please save.');
         }
     }
 
-    /**
-     * Return the timestamp in the following format: 2020-04-06 10:25:50.123.
-     */
-    private getCurrentTime(): string {
-        let time = new Date();
-        time = new Date(time.getTime() - (time.getTimezoneOffset() * 60000));
-        return time.toISOString().substring(0, 23).replace(/[T]/g, ' ');
-    }
-
     /**
      * Save the current context of the active editor.
      */
     public async saveActiveEditorContext(document: vscode.TextDocument): Promise<void> {
         const timestamp = this.getCurrentTime();
         const timestampForFileName = timestamp.replace(/[-:. ]/g, '');
-        const timestampForProperty = timestamp.substring(0, 19);
         const fileFullPath = path.parse(document.fileName);
         const historyFileName = `${fileFullPath.name}_${timestampForFileName.substring(0, 14)}_${timestampForFileName.substring(14, 17)}${fileFullPath.ext}`;
 
         if (vscode.workspace.getWorkspaceFolder(document.uri) !== undefined) {
-            const workspaceFolderPath: string = vscode.workspace.getWorkspaceFolder(document.uri)!.uri.fsPath;
+            const localHistoryFolderPath = path.join(os.homedir(), '.local-history');
+            const hashedEditorPath = checksum(document.fileName);
+            const hashedFolderPath = path.join(localHistoryFolderPath, hashedEditorPath);
 
-            const historyFolderPath = path.join(workspaceFolderPath, '.local-history');
-
-            // Create .local-history folder if it does not exist.
-            if (!fs.existsSync(historyFolderPath)) {
-                fs.mkdirSync(historyFolderPath, { recursive: true });
+            // Create a folder (and all the parent folders) for storing all the local history file for the active editor.
+            if (!fs.existsSync(hashedFolderPath)) {
+                fs.mkdirSync(hashedFolderPath, { recursive: true });
             }
 
-            // Copy the content of the current active editor.
-            const historyFilePath = path.join(workspaceFolderPath, '.local-history', historyFileName);
-            const data: HistoryFileProperties = {
-                fileName: historyFileName,
-                timestamp: timestampForProperty,
-                uri: historyFilePath,
-                parentFileName: fileFullPath.base
-            };
-            this.historyFiles.unshift(data);
```

Comment:
historyFiles should be updated with the content.

Reply:
Instead of all the local history file, `historyFiles` now stores the history files for the active editor. We load history files from user's home each time we trigger `View All for Active Editor`.